bitnami/argo-cd/templates/application-controller/sts-dpl.yaml

```diff
@@ -0,0 +1,224 @@
+                    info "Connected to the Redis instance"
+                fi
+          env:
+            - name: REDISCLI_AUTH
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "argocd.redis.secretName" . }}
+                  key: {{ include "argocd.redis.secretPasswordKey" . }}
+        {{- if .Values.controller.initContainers }}
+        {{- include "common.tplvalues.render" (dict "value" .Values.controller.initContainers "context" $) | nindent 8 }}
+        {{- end }}
+      containers:
+        - name: controller
+          image: {{ include "argocd.application-controller.image" . }}
+          imagePullPolicy: {{ .Values.controller.image.pullPolicy }}
+          {{- if .Values.controller.lifecycleHooks }}
+          lifecycle: {{- include "common.tplvalues.render" (dict "value" .Values.controller.lifecycleHooks "context" $) | nindent 12 }}
+          {{- end }}
+          {{- if .Values.controller.containerSecurityContext.enabled }}
+          securityContext: {{- omit .Values.controller.containerSecurityContext "enabled" | toYaml | nindent 12 }}
+          {{- end }}
+          {{- if .Values.controller.command }}
+          command: {{- include "common.tplvalues.render" (dict "value" .Values.controller.command "context" $) | nindent 12 }}
+          {{- else }}
+          command:
+            - argocd-application-controller
+          {{- end }}
+          {{- if .Values.controller.args }}
+          args: {{- include "common.tplvalues.render" (dict "value" .Values.controller.args "context" $) | nindent 12 }}
+          {{- else }}
+          args:
+            - --status-processors
+            - {{ .Values.controller.defaultArgs.statusProcessors | quote }}
+            - --operation-processors
+            - {{ .Values.controller.defaultArgs.operationProcessors | quote }}
+            - --app-resync
+            - {{ .Values.controller.defaultArgs.appResyncPeriod | quote }}
+            - --self-heal-timeout-seconds
+            - {{ .Values.controller.defaultArgs.selfHealTimeout | quote }}
+            - --repo-server
+            - {{ include "argocd.application-controller" . }}:{{ .Values.repoServer.service.port }}
+            - --logformat
+            - {{ .Values.controller.logFormat }}
+            - --loglevel
+            - {{ .Values.controller.logLevel }}
+            # TODO(miguelaeh): Test the chart using redis sentinel enabled: https://github.com/argoproj/argo-cd/blob/2a410187565e15633b6f2a8c8d8da22cf02b257d/util/cache/cache.go#L40
+            - --redis
+            - {{ include "argocd.redisHost" . }}:{{ include "argocd.redisPort" . }}
+          {{- if .Values.controller.extraArgs }}
+          {{- include "common.tplvalues.render" (dict "value" .Values.controller.extraArgs "context" $) | nindent 8 }}
+          {{- end }}
+          {{- end }}
+          ports:
+            - name: controller
+              containerPort: {{ .Values.controller.ports.controller }}
```

Comment:
We are using in other charts the `containerPort` suffix, so instead of `ports.controller` and `ports.metrics` we would use `containerPort` (as the controller is the main one) and `metricsContainerPort`

Reply:
I don't think that's appropriate here, this chart has a lot of ports in other components, you can for example filter them from the README:
```
109:| `controller.ports.controller`                            | Argo CD application controller port number                                                           | `8082`               |
110:| `controller.ports.metrics`                               | Argo CD application controller metrics port number                                                   | `8082`               |
240:| `server.ports.http`                                  | Argo CD server HTTP container port                                                               | `8080`                   |
241:| `server.ports.https`                                 | Argo CD server HTTPS container port                                                              | `8443`                   |
242:| `server.ports.metrics`                               | Argo CD server metrics container port                                                            | `8083`                   |
244:| `server.service.ports.http`                          | HTTP port for the gRPC ingress when enabled                                                      | `80`                     |
245:| `server.service.ports.https`                         | HTTPS port for the gRPC ingress when enabled                                                     | `443`                    |
246:| `server.service.nodePorts.http`                      | Node port for HTTP                                                                               | `nil`                    |
247:| `server.service.nodePorts.https`                     | Node port for HTTPS                                                                              | `nil`                    |
320:| `repoServer.ports.repoServer`                            | Container port for Argo CD repo server                                                               | `8081`               |
321:| `repoServer.ports.metrics`                               | Metrics port for Argo CD repo server                                                                 | `nil`                |
398:| `dex.service.ports.http`                          | Dex HTTP service port                                                                         | `5556`                |
399:| `dex.service.ports.grpc`                          | Dex grpc service port                                                                         | `5557`                |
400:| `dex.service.nodePorts.http`                      | HTTP node port for the Dex service                                                            | `nil`                 |
401:| `dex.service.nodePorts.grpc`                      | gRPC node port for the Dex service                                                            | `nil`                 |
406:| `dex.ports.http`                                  | Dex container HTTP port                                                                       | `5556`                |
407:| `dex.ports.grpc`                                  | Dex gRPC port                                                                                 | `5557`                |
408:| `dex.ports.metrics`                               | Dex metrics port                                                                              | `5558`                |
```

I don't think adding a `containerPort` suffix to all of them adds any value since it is already separated from the service ports (that will go into the service section)

Reply:
It's more about consistency and having similar UX with all our charts. We need to come up with something to standardize but, looking at the rest of the charts, let's do al least the following

```
controller.containerPorts.controller
controller.containerPorts.metrics
server.containerPorts....

```

and so on